src/main/java/core/basesyntax/handler/BalanceOperationHandler.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax.handler;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.storage.FruitStorage;
+
+public class BalanceOperationHandler implements OperationHandler {
+    private final FruitStorage fruitStorage;
+
+    public BalanceOperationHandler(FruitStorage fruitStorage) {
+        this.fruitStorage = fruitStorage;
+    }
+
+    @Override
+    public void apply(FruitTransaction transaction) {
+        fruitStorage.setFruit(transaction.getFruit(), transaction.getQuantity());
```

Comment:
Ensure that the `setFruit` method in `FruitStorage` is the correct method for handling the balance operation. The balance operation typically sets the quantity of fruit to a specific value, so the method name should reflect this action clearly (e.g., `setBalance`). If `setFruit` is indeed the correct method, consider renaming it to better convey its purpose.